xbout/geometries.py

```diff
@@ -418,3 +423,11 @@ def add_s_alpha_geometry_coords(ds, *, coordinates=None, grid=None):
         del ds["hthe"]
 
     return ds
+
+
+@register_geometry("fci")
+def add_fci_geometry_coords(ds, *, coordinates=None, grid=None):
+    assert coordinates is None, "Not implemented"
+    ds = _add_vars_from_grid(ds, grid, ["R", "Z"])
+    ds = ds.set_coords(("R", "Z"))
```

Comment:
```suggestion
    ds = ds.set_coords(("R", "Z"))
    ds = _create_single_region(ds, periodic_y=True)
```
Creating a 'region' here should allow more methods to work - that was why I added this function originally, but I don't think there's much at the moment that's actually useful. It should be possible to use for example `ds["n"].isel(y=0).bout.pcolormesh()` to get an R-Z plot, but there's not a lot of advantage of that over `ds["n"].isel(y=0).plot(x="R", y="Z")`, although we could add functionality (see 1. below)

It might be useful eventually to:
1.  extend `Region` to handle the z-direction as well.
    For example, if the z-direction is periodic and you want to plot an x-z plane (which is probably common with FCI?) you might end up with a white gap between the end and beginning of the z-grid (although I think the latest versions (3.4.x) of matplotlib are better for this). One solution, which we use when plotting x-y planes in conventional tokamak geometry, is to add a guard cell at any branch cuts - it should be a straightforward extension to allow the `Region` functionality under `BoutDataArray.from_region()` to get z-guard cells (from periodicity), and to allow the geometry to know whether z should be treated as periodic or not.
2. extend `BoutDataArray.ddy()` to recognise and handle FCI data.